components/NewStoryForm/index.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import Image from "next/image";
+
+const Form = styled.form`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  margin: 2rem;
+`;
+
+const Input = styled.input`
+  margin-bottom: 1rem;
+  padding: 0.5rem;
+  font-size: 1rem;
+`;
+
+const TextArea = styled.textarea`
+  margin-bottom: 1rem;
+  padding: 0.5rem;
+  font-size: 1rem;
+  width: 90%;
+  height: 50vh;
+  resize: none;
+`;
+
+const Button = styled.button`
+  margin-top: 1rem;
+  padding: 0.5rem 1rem;
+  font-size: 1rem;
+  background-color: #333;
+  color: #fff;
+  border: none;
+  border-radius: 4px;
+  cursor: pointer;
+`;
+
+export default function NewStoryForm({ onSubmit, setStories }) {
+  const [title, setTitle] = useState("");
+  const [coverImage, setCoverImage] = useState(null);
+  const [textContent, setTextContent] = useState("");
+
+  function handleTitleChange(event) {
+    setTitle(event.target.value);
+  }
+
+  function handleImageChange(event) {
+    setCoverImage(event.target.files[0]);
+  }
+
+  function handleTextChange(event) {
+    setTextContent(event.target.value);
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const storiesFromLocalStorage =
+      JSON.parse(localStorage.getItem("stories")) || [];
```

Comment:
I like your usage of local storage. In the future it might be tricky to connect state and local storage correclty. If so think about using the local-storage-state hook [(see our session handout for local storage)](https://github.com/neuefische/hh-web-23-2/blob/main/sessions/react-with-local-storage/react-with-local-storage.md#react-with-local-storage-1).